airbyte-integrations/connectors/source-genesys/unit_tests/test_analytics_utils.py

```diff
@@ -0,0 +1,115 @@
+import hashlib
+from datetime import datetime
+
+from source_genesys.analytics_utils import (
+    generate_query, 
+    create_surrogate_key, 
+    str_timestamp_to_datetime, 
+    split_utc_timestamp_interval,
+    parse_analytics_records
+)
+
+
+def test_generate_query():
+    window_date_str = "2023-05-17"
+    metrics = ["metric1", "metric2"]
+    expected_query = {
+        "interval": "2023-05-17T00:00:00.000Z/2023-05-17T23:59:59.000Z",
+        "metrics": metrics
+    }
+
+    result = generate_query(window_date_str, metrics)
+
+    assert result == expected_query
+
+
+def test_create_surrogate_key():
+    key1 = "abc"
+    key2 = 123
+    key3 = "xyz"
```

Comment:
what if key is null or "" ?

Reply:
Good qs @jojoms711 
So we always expect non-null values for the keys we pass to the create_surrogate_key() function in this use case (otherwise the connection will fail). 
But to answer your question, a null value when passed to this function will be disregarded.
I have updated the unit test here to prove this:
`hash(key1,key2,key3) == create_surrogate_key(key1, key2, key3, empty_key_val)`